src/kofta_metadata.erl

```diff
@@ -254,16 +276,36 @@ make_request(State) ->
 
     TopicNames = dict:fetch_keys(ClientDict),
 
-    NewState = State#st{clients=dict:new(), last_batch=now()},
+    NewState = State#st{
+        clients=dict:new(),
+        broker_clients=dict:new(),
+        last_batch=now()
+    },
 
     case do_request(TopicNames, Host, Port) of
-        {ok, Topics} ->
-            lists:map(fun(Topic) ->
-                Clients = dict:fetch(Topic#topic.name, ClientDict),
-                lists:map(fun(Client) ->
-                    gen_fsm:reply(Client, {ok, Topic})
-                end, Clients)
-            end, Topics),
+        {ok, Brokers, Topics} ->
+            case dict:size(ClientDict) of
+                0 ->
```

Comment:
Well, it's going to be an issue either way, I think, because your `dict:fetch` call is going to be throwing exceptions for topic no one is waiting for (I think). I'll think about it more after I eat some food.